python/paddle/v2/framework/tests/op_test.py

```diff
@@ -5,175 +5,313 @@
+                        out_var_tensor = cls.get_tensor_by_name(scope,
+                                                                sub_var_name)
+                        data = np.ones(out_var_tensor.shape(), dtype=np.float64)
+                        cls.create_variable(
+                            scope,
+                            cls.grad_var_name(sub_var_name),
+                            value=data,
+                            place=place)
+                else:
+                    out_var_tensor = cls.get_tensor_by_name(scope, out_var_name)
+                    data = np.ones(out_var_tensor.shape(), np.float64)
+                    cls.create_variable(
+                        scope,
+                        cls.grad_var_name(out_var_name),
+                        value=data,
+                        place=place)
+
+        backward_op.infer_shape(scope)
+        backward_op.run(scope, ctx)
+        out = np.array(cls.get_tensor_by_name(scope, grad_name))
+        return out
+
+    @classmethod
+    def get_out_var_shapes(cls, scope, op, outputs):
+        out_var_shapes = []
+        for out_var_name, is_dup in Operator.get_op_outputs(op.type()):
+            if out_var_name in outputs:
+                if is_dup:
+                    sub_vars = outputs[out_var_name]
+                    for sub_var_name, _ in sub_vars:
+                        out_var_tensor = cls.get_tensor_by_name(scope,
+                                                                sub_var_name)
+                        out_var_shapes.append(
+                            (sub_var_name, out_var_tensor.shape()))
+                else:
+                    out_var_tensor = cls.get_tensor_by_name(scope, out_var_name)
+                    out_var_shapes.append(
+                        (out_var_name, out_var_tensor.shape()))
+        return out_var_shapes
+
+    # TODO(pkuyym) should pass output_names not outputs
+    @classmethod
+    def get_jacobian_analytic_grads(cls,
+                                    scope,
+                                    op,
+                                    inputs,
+                                    outputs,
+                                    grad_name,
+                                    place,
+                                    no_grad_set=None):
+        # only run forward one time
+        ctx = core.DeviceContext.create(place)
+        cls.feed_input(scope, op, inputs, place)
+        op.infer_shape(scope)
+        op.run(scope, ctx)
+
+        # get shape for each outputs, may pass by outside
+        out_var_shapes = cls.get_out_var_shapes(scope, op, outputs)
+        accum_size = np.zeros((len(out_var_shapes)), dtype=np.int32)
+        var_shape_idx = {}
+        for i in xrange(len(out_var_shapes)):
+            accum_size[i] = cls.dim_to_size(out_var_shapes[i][1]) + \
+                    (accum_size[i - 1] if i > 0 else 0)
+            var_shape_idx[out_var_shapes[i][0]] = i
```

Comment:
Change `out_var_shapes[i][0]` to `out_var_shapes[i][0].encode('utf-8')` to keep consistent with line 264.

Reply:
change ```tensor_name.encode('utf-8')``` to ```tensor_name```